chromadb/test/property/test_clean_log.py

```diff
@@ -0,0 +1,55 @@
+from typing import cast
+from overrides import overrides
+from chromadb.api import ServerAPI
+from chromadb.db.base import get_sql
+from chromadb.db.impl.sqlite import SqliteDB
+from chromadb.ingest import Producer
+from pypika import Table, functions
+from hypothesis.stateful import (
+    run_state_machine_as_test,
+    invariant,
+)
+
+from chromadb.test.property.test_embeddings import (
+    EmbeddingStateMachineBase,
+    EmbeddingStateMachineStates,
+)
+
+
+def count_embedding_queue_rows(sqlite: SqliteDB) -> int:
+    t = Table("embeddings_queue")
+    q = sqlite.querybuilder().from_(t).select(functions.Count(t.seq_id))
+
+    with sqlite.tx() as cur:
+        sql, params = get_sql(q, sqlite.parameter_format())
+        result = cur.execute(sql, params)
+        return cast(int, result.fetchone()[0])
+
+
+class LogCleanEmbeddingStateMachine(EmbeddingStateMachineBase):
+    has_collection_mutated = False
+
+    @invariant()
+    def log_empty_after_cleaning(self) -> None:
+        producer = self.api._system.instance(Producer)
+        sqlite = self.api._system.instance(SqliteDB)
+
+        producer.clean_log(self.collection.id)
```

Comment:
I'm not sure if creating a side effect like this inside an invariant is the recommended approach; I didn't see another way

Reply:
todo: should be rule?

Reply:
Does `empty` make sense when there are unsaved HNSW embeddings e.g. hnsw:sync_threshold_size not been reached?